rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -594,6 +596,23 @@ private void saveSvpFundTxHashSigned() {
         repository.addStorageBytes(contractAddress, SVP_FUND_TX_HASH_SIGNED.getKey(), data);
     }
 
+    public void setSvpSpendTxHashUnsigned(Sha256Hash hash) {
+        this.svpSpendTxHashUnsigned = hash;
+        this.isSvpSpendTxHashUnsignedSet = true;
+    }
+
+    private void saveSvpSpendTxHashUnsigned() {
+        if (!activations.isActive(RSKIP419) || !isSvpSpendTxHashUnsignedSet) {
+            return;
+        }
```

Comment:
I am not sure about the purpose of the `isSvpSpendTxHashUnsignedSet` variable. From my understanding, we are never setting this variable back to false once we store it. So in a subsequent iteration this condition will always pass with whatever hash that is set. Is this expected?

Also shouldn't the if condition be using an AND operator instead of an OR operator? 

Maybe I am missing something here.